lib/get-transitive-data.js

```diff
@@ -61,62 +79,38 @@ export default function getTransitiveData ({
     log(e.message)
   }
 
-  output.places.push({
+  places.push({
     place_id: 'to',
     place_name: to.name,
     place_lat: toCoord.lat || pixelToLat(query.north + to.y, query.zoom),
     place_lon: toCoord.lon || pixelToLon(query.west + to.x, query.zoom)
   })
 
-  // find relevant paths at each minute
-  // this array stores the destination stop and the path index in that stop
-  paths = paths.filter(p => p !== undefined)
-
-  paths.sort((p1, p2) => {
-    const v0 = p1[0] - p2[0] // sort first by stop ID
-    if (v0 !== 0) return v0
-    else return p1[1] - p2[1] // then by path index
-  })
-
-  const npaths = paths.length
-
-  // get (up to) five most common paths
-  paths = paths.reduce((out, path, index) => {
-    if (index === 0) out.unshift({ count: 1, path })
-    else if (out[0].path[0] === path[0] && out[0].path[1] === path[1]) out[0].count++
-    else out.unshift({ count: 1, path })
-    return out
-  }, [])
-
-  // sort by frequency
-  paths.sort((p1, p2) => p2.count - p1.count)
-
-  log(`${paths.length} unique paths`)
-
-  if (paths.length > 5) {
-    log(`eliminated ${paths.length - 5} paths with frequency less than or equal to ${paths[5].count} / ${npaths} `)
+  // function to convert pattern ID to route ID
+  function getRouteIndexFromPatternId (pattern: string | number): number {
+    const pat = transitiveNetwork.patterns.find(p => p.pattern_id === String(pattern))
+    return transitiveNetwork.routes.findIndex(r => r.route_id === pat.route_id)
   }
 
-  paths = paths.slice(0, 5).map(p => p.path)
-
-  // uniquify the paths
-  // always return first path, and don't return paths the same as the previous
-  paths = paths.filter((p, idx, arr) =>
-    idx === 0 || p[0] !== arr[idx - 1][0] || p[1] !== arr[idx - 1][1]).map(p =>
-      getPath({log, pathDescriptor: p, origin}))
-
-  paths = filterPaths({log, transitiveNetwork, paths})
+  // get the most common, unique paths
+  const commonPaths = getCommonPaths({
+    getRouteIndexFromPatternId,
+    log,
+    origin,
+    paths: paths.filter((p) => !!p)
+  })
 
-  for (let pidx = 0; pidx < paths.length; pidx++) {
+  for (let pidx = 0; pidx < commonPaths.length; pidx++) {
     const journey = {
       journey_id: pidx,
       journey_name: pidx,
       segments: []
     }
 
-    const path = paths[pidx]
+    const path = commonPaths[pidx]
 
-    // bail if the path-finding algorithm gave up, this is a bug but we'd rather log a javascript error than crash the tab
+    // bail if the path-finding algorithm gave up, this is a bug but we'd rather
+    // log a javascript error than crash the tab (??)
```

Comment:
i.e. we don't want to have this throw a 'cannot read property whatever of undefined' when it tries to read a property of the (null) path.